cns/client/client.go

```diff
@@ -411,3 +411,105 @@ func (c *Client) GetHTTPServiceData(ctx context.Context) (*restserver.GetHTTPSer
 
 	return &resp, nil
 }
+
+// DeleteNetworkContainer destroys the requested network container matching the
+// provided ID.
+func (c *Client) DeleteNetworkContainer(ncID string) error {
+	// the network container ID is required by the API, so ensure that we have
+	// one before we even make the request
+	if ncID == "" {
+		return errors.New("no network container ID provided")
+	}
+
+	// build the request
+	dncr := cns.DeleteNetworkContainerRequest{
+		NetworkContainerid: ncID,
+	}
+	body, err := json.Marshal(dncr)
+	if err != nil {
+		return errors.Wrap(err, "encoding request body")
+	}
+	u := c.routes[cns.DeleteNetworkContainer]
+	req, err := http.NewRequest(http.MethodPost, u.String(), bytes.NewReader(body))
+	if err != nil {
+		return errors.Wrap(err, "building HTTP request")
+	}
+
+	// submit the request
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "sending HTTP request")
+	}
+	defer resp.Body.Close()
+
+	// decode the response
+	var out cns.DeleteNetworkContainerResponse
+	err = json.NewDecoder(resp.Body).Decode(&out)
+	if err != nil {
+		return errors.Wrap(err, "decoding response as JSON")
+	}
+
+	// if a non-zero response code was received from CNS, it means something went
+	// wrong and it should be surfaced to the caller as an error
+	if out.Response.ReturnCode != 0 {
+		return errors.New(out.Response.Message)
```

Comment:
Would a client want to handle errors differently for different error codes? I see other methods doing something similar but if we are going to start consuming this from DNC, I would like to leave this as open for future compatibility and return the actual API contract struct. Probably create new methods for all the methods that we will use from DNC.

Reply:
I think it is likely that a client would want to handle errors differently (notably retries for temporary errors). I still think the better way to do it is by having a `TemporaryError` like I did in the NMAgent client (this is the same idea as Dave Cheney's ["don't just check errors, handle them gracefully"](https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully)).

I like this because the knowledge about what constitutes a temporary error (i.e. what do the error codes mean) is in exactly one place: the `IsTemporary()` method (or equivalent for other classes of error).